Mathlib/Analysis/SpecialFunctions/Pow/Real.lean

```diff
@@ -513,6 +513,31 @@ lemma rpow_mul_intCast (hx : 0 ≤ x) (y : ℝ) (n : ℤ) : x ^ (y * n) = (x ^ y
 /-! Note: lemmas about `(∏ i in s, f i ^ r)` such as `Real.finset_prod_rpow` are proved
 in `Mathlib/Analysis/SpecialFunctions/Pow/NNReal.lean` instead. -/
 
+/-- `rpow` version of `Finset.prod_pow_eq_pow_sum`. -/
+theorem finset_prod_rpow_eq_rpow_sum
+    {ι} (s : Finset ι) (f : ι → ℝ) (hs : ∀ i ∈ s, 0 ≤ f i) (r : ℝ) (hr : 0 ≤ r) :
+    ∏ i in s, r ^ f i = r ^ (∑ i in s, f i) := by
+  refine' Finset.cons_induction_on s
+    (p := fun s ↦ (∀ i ∈ s, 0 ≤ f i) → ∏ i in s, r ^ f i = r ^ ∑ i in s, f i) _ _ hs
+  · simp only [Finset.prod_empty, sum_empty, rpow_zero, implies_true]
+  · intros a s ha ih hs
+    have h : ∀ i ∈ s, 0 ≤ f i := fun i hi ↦ hs i (Finset.mem_cons.mpr (Or.inr hi))
+    rw [Finset.sum_cons, Finset.prod_cons, ih h, rpow_add_of_nonneg hr ?_ (Finset.sum_nonneg h)]
```

Comment:
Just so you know: using dot notation, you can write this in a shorter way. (You don't have to, though!)
`rw [s.sum_cons, s.prod_cons, ih h, rpow_add_of_nonneg hr ?_ (s.sum_nonneg h)]`

Reply:
Thank you. Good to know.